lib/using_restricted_array.rb

```diff
@@ -1,65 +1,116 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
+
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: o(n)
+# Space complexity: o(1)
 def length(array)
-  raise NotImplementedError
+  length = 0
+
+  until array[length] == nil
+    length += 1
+  end
+
+  return length
 end
 
-# Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# # Prints each integer values in the array
+# # Time complexity: o(n)
+# # Space complexity: o(1)
 def print_array(array)
-  raise NotImplementedError
+  counter = 0
+  until array[counter] == nil
+    puts array[counter]
+  end
 end
 
-# For an unsorted array, searches for 'value_to_find'.
-# Returns true if found, false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# # For an unsorted array, searches for 'value_to_find'.
+# # Returns true if found, false otherwise.
+# # Time complexity: o(n)
+# # Space complexity: o(1)
 def search(array, length, value_to_find)
-  raise NotImplementedError
+  length.times do |i|
+    if array[i] == value_to_find
+      return true
+    end
+  end
+  return false
 end
 
-# Finds and returns the largest integer value the array
-# Assumes that the array is not sorted.
-# Time complexity: ?
-# Space complexity: ?
+# # Finds and returns the largest integer value the array
+# # Assumes that the array is not sorted.
+# # Time complexity: o(n)
+# # Space complexity: o(1)
 def find_largest(array, length)
-  raise NotImplementedError
+  max = 0
```

Comment:
What if the array only holds negative numbers?

Better to set `max = array[0]`